services/web-server/src/main.js

```diff
@@ -118,13 +119,30 @@ const load = loader(
     },
 
     httpServer: {
-      requires: ['app', 'schema', 'context'],
-      setup: ({ app, schema, context }) => {
+      requires: ['app', 'schema', 'context', 'monitor'],
+      setup: ({ app, schema, context, monitor }) => {
+        const logger = monitor.childMonitor('httpServer');
+        const monitorPlugin = {
+          // Fires whenever a GraphQL request is received from a client.
+          requestDidStart(requestContext) {
+            const { request } = requestContext;
+
+            if (request.operationName !== "IntrospectionQuery") {
+              logger.log.requestReceived({
+                operationName: request.operationName,
+                variables: request.variables,
+                query: request.query,
+                requestId: slugid.v4(),
```

Comment:
Is there a way to thread this value through so that we can access it when making REST API requests?  (We can figure out how to pass it to API requests in a followup)

Reply:
Yes. We can have the UI send a request-id header that each graphql loader would have access to. Something like:

```diff
diff --git a/services/web-server/src/loaders/workerManager.js b/services/web-server/src/loaders/workerManager.js
index 62b031286..a7973c3f9 100644
--- a/services/web-server/src/loaders/workerManager.js
+++ b/services/web-server/src/loaders/workerManager.js
@@ -2,7 +2,8 @@ const DataLoader = require('dataloader');
 const sift = require('../utils/sift');
 const ConnectionLoader = require('../ConnectionLoader');
 
-module.exports = ({ workerManager }) => {
+module.exports = ({ workerManager }, isAuthed, rootUrl, monitor, strategies, cfg) => {
+  console.log('loader has acccess to request id: ', cfg.headers['request-id']);
   const WorkerManagerWorkerPoolSummaries = new DataLoader(queries =>
     Promise.all(
       queries.map(async ({ filter }) => {
diff --git a/ui/src/App/index.jsx b/ui/src/App/index.jsx
index 1ec445fb3..dddae58ac 100644
--- a/ui/src/App/index.jsx
+++ b/ui/src/App/index.jsx
@@ -29,6 +29,7 @@ import theme from '../theme';
 import introspectionQueryResultData from '../fragments/fragmentTypes.json';
 import { route } from '../utils/prop-types';
 import AuthController from '../auth/AuthController';
+import { v4 } from '../utils/slugid';
 
 const absoluteUrl = (url, overrides = {}) =>
   Object.assign(new URL(url, window.location), overrides).toString();
@@ -103,6 +104,7 @@ export default class App extends Component {
           headers: {
             ...headers,
             Authorization: `Bearer ${btoa(JSON.stringify(user.credentials))}`,
+            'request-id': v4(),
           },
         };
       }),
```

Let me know if this sounds good to you. I could open another PR that sends the request-id to each REST API request.

Reply:
I think it would be better to generate the request-id here in requestDidStart, and then pass it along to loaders (rather than trusting the HTTP client to send an appropriately-random requestId)

Reply:
That works too.

Reply:
I didn't know of a way to send the request id to the context when the request id is generated in the plugin (i.e., requestdidStart). Instead, I started generating the requestId inside the context in `createContext.js` which gets called on each request.